### PR TITLE
API v1 is the default API

### PR DESCRIPTION
**Pull request by ouziel-slama**

### CODE REVIEW

File: release-notes/release-notes-v10.1.2.md [hunk 1]
```diff
@@ -21,7 +14,7 @@ There is a [migration guide](https://docs.counterparty.io/docs/advanced/api-v2/v
 
 
 ## Codebase
-* Release API v2; deprecate API v1
+* Release API v2;
 * Have API return `ready` if the last block is less than one minute old
```
Comment (adamkrellenstein) [inline release-notes/release-notes-v10.1.2.md:22]: Does this apply to API v1 as well?

File: counterparty-core/counterpartycore/lib/config.py [hunk 2]
```diff
@@ -51,13 +51,13 @@
 FULL_APP_NAME = "Counterparty Core"
 LOGGER_NAME = APP_NAME
 
-DEFAULT_API_PORT_REGTEST = 24000
-DEFAULT_API_PORT_TESTNET = 14000
-DEFAULT_API_PORT = 4000
+DEFAULT_API_PORT_REGTEST = 24200
+DEFAULT_API_PORT_TESTNET = 14200
+DEFAULT_API_PORT = 4200
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/config.py:56]: Ach sorry I meant `*4100` not `*4200`.

File: release-notes/release-notes-v10.1.2.md (hunk 1 above)
Comment (ouziel-slama) [reply release-notes/release-notes-v10.1.2.md:22]: yep

**Comment by ouziel-slama** [comment]
- [x] puts v2 API behind `/v2/`
- [x] redirect `/`, `/rpc/`, `/api/` and `/v1/` to API v1
- [x] disables basic authentication if no password is provided with `--api-password`.
- [x] API v1 is always accessible without redirection on port *4100 and always with rpc:rpc by default.

File: counterparty-core/counterpartycore/test/fixtures/api_v2_fixtures.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-    "http://api:api@localhost:10009/blocks": {
+    "http://rpc:rpc@localhost:10009/v2/blocks": {
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/test/fixtures/api_v2_fixtures.json:2]: We could remove the basic auth here, no?

**Review by adamkrellenstein — APPROVED** [review]

**Comment by adamkrellenstein** [comment]
🎉

File: counterparty-core/counterpartycore/test/fixtures/api_v2_fixtures.json (hunk 3 above)
Comment (ouziel-slama) [reply counterparty-core/counterpartycore/test/fixtures/api_v2_fixtures.json]: yep. done.

**Pull request merged**